google/auth/downscoped.py

```diff
@@ -467,7 +471,10 @@ def __init__(
         self._source_credentials = source_credentials
         self._credential_access_boundary = credential_access_boundary
         self._quota_project_id = quota_project_id
-        self._sts_client = sts.Client(_STS_TOKEN_URL)
+        self._universe_domain = universe_domain or credentials.DEFAULT_UNIVERSE_DOMAIN
```

Comment:
Can you check that the universe domain matches the one in source creds?

Reply:
It is covered in the test L#507
```
        assert credentials.universe_domain == DEFAULT_UNIVERSE_DOMAIN
```

Reply:
What I mean here is if you pass source creds with universe domain A, but pass universe domain B here. We should error out in this case

Reply:
I think the test supposed to be:
1. Implicit default. Pass nothing for universe domain and validate the universe domain as the default.
2. Explicitly pass None. And valid the universe domain is equals to default.
3. Explicitly pass a universe domain A, and validate the universe domain equals to A

Now working on the test above. Any missing validation?